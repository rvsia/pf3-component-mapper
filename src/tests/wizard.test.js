import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import MockFieldProvider from './mock-field-provider';
import Wizard from '../form-fields/wizzard/wizzard';
import FormRenderer from '@data-driven-forms/react-form-renderer';
import { componentTypes, validatorTypes } from '@data-driven-forms/react-form-renderer';
import { formFieldsMapper, layoutMapper } from '../';

describe('<Wizard />', () => {
  const cancelSpy = jest.fn();
  const submitSpy = jest.fn();

  const props = {
    name: 'Wizard',
    FieldProvider: MockFieldProvider,
    fields: [
      {
        title: 'Step 1',
        name: 'step-1',
        stepKey: 1,
        nextStep: 'step-2',
        fields: [],
      },
      {
        title: 'Step 2',
        name: 'step-2',
        stepKey: 'step-2',
        fields: [],
      },
    ],
    formOptions: {
      onSubmit: submitSpy,
      onCancel: cancelSpy,
      valid: true,
      getState: () => ({ values: { a: 10 }}),
      submit: () => {},
    },
  };

  const conditionalProps = {
    ...props,
    fields: [
      {
        title: 'Step 1',
        name: 'step-1',
        stepKey: 1,
        nextStep: {
          when: 'step',
          stepMapper: {
            step: 'step-2',
          },
        },
        fields: [],
      },
      {
        title: 'Step 2',
        name: 'step-2',
        stepKey: 'step-2',
        fields: [],
      },
    ],
  };

  const conditionalSchema = {
    fields: [{
      component: componentTypes.WIZARD,
      name: 'wizzard',
      fields: [{
        name: 'step-1',
        stepKey: 1,
        nextStep: {
          when: 'source-type',
          stepMapper: {
            aws: 'aws-step',
            google: 'google-step',
          },
        },
        fields: [{
          component: componentTypes.TEXTAREA_FIELD,
          name: 'source-name',
          type: 'text',
          label: 'Source name',
        }, {
          component: componentTypes.SELECT_COMPONENT,
          name: 'source-type',
          label: 'Source type',
          options: [{
            label: 'Please Choose',
          }, {
            value: 'aws',
            label: 'Aws',
          }, {
            value: 'google',
            label: 'Google',
          }],
          validate: [{
            type: validatorTypes.REQUIRED,
          }],
        }],
      }, {
        name: 'step-2',
        stepKey: 'aws-step',
        fields: [{
          component: componentTypes.TEXT_FIELD,
          name: 'aws-field',
          label: 'Aws field part',
        }],
      }, {
        stepKey: 'google-step',
        name: 'step-3',
        fields: [{
          component: componentTypes.TEXT_FIELD,
          name: 'google-field',
          label: 'Google field part',
        }],
      }],
    }],
  };

  afterEach(() => {
    cancelSpy.mockReset();
    submitSpy.mockReset();
  });

  it('should render Wizard correctly', () => {
    const wrapper = mount(
      <Wizard { ...props } />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Wizard with conditional steps correctly', () => {
    const wrapper = mount(
      <Wizard { ...conditionalProps } />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Wizard with title correctly', () => {
    const wrapper = mount(
      <Wizard { ...props } title={ 'Wizard title' }/>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Wizard with stepsInfo correctly', () => {
    const wrapper = mount(
      <Wizard { ...props } stepsInfo={ [{ title: 'step1' }, { title: 'step2' }] }/>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call cancel', () => {
    const wrapper = mount(
      <Wizard { ...props } />
    );
    wrapper.find('button').first().simulate('click');
    wrapper.update();
    expect(cancelSpy).toHaveBeenCalled();
  });

  it('should step when clicked on next button', () => {
    const wrapper = mount(
      <Wizard { ...props } />
    );
    expect(wrapper.instance().state.activeStep).toEqual(1);

    const nextButton = wrapper.find('button').last();
    nextButton.simulate('click');

    expect(wrapper.instance().state.activeStep).toEqual('step-2');
  });

  it('should not step when clicked on button with false valid', () => {
    const wrapper = mount(
      <Wizard { ...props } formOptions={{ ...props.formOptions, valid: false }}  />
    );
    expect(wrapper.instance().state.activeStep).toEqual(1);

    const nextButton = wrapper.find('button').last();
    nextButton.simulate('click');

    expect(wrapper.instance().state.activeStep).toEqual(1);
  });

  it('should submit when clicked on next button 2x', () => {
    const wrapper = mount(
      <Wizard { ...props } />
    );
    expect(submitSpy).not.toHaveBeenCalled();

    let nextButton = wrapper.find('button').last();
    nextButton.simulate('click');
    nextButton = wrapper.find('button').last();
    nextButton.simulate('click');

    expect(submitSpy).toHaveBeenCalled();
  });

  it('should stepBack when clicked on back button', () => {
    const wrapper = mount(
      <Wizard { ...props } />
    );
    expect(wrapper.instance().state.activeStep).toEqual(1);

    const nextButton = wrapper.find('button').last();
    nextButton.simulate('click');

    expect(wrapper.instance().state.activeStep).toEqual('step-2');

    const backButton = wrapper.find('button').at(1);
    backButton.simulate('click');

    expect(wrapper.instance().state.activeStep).toEqual(1);
  });

  it('should step to google-step when clicked on next button in conditional schema', () => {
    const wrapper = mount(
      <FormRenderer
        schema={ conditionalSchema }
        formFieldsMapper={ formFieldsMapper }
        layoutMapper={ layoutMapper }
        onCancel={ () => {} }
        showFormControls={ false }
        onSubmit={ jest.fn() }
        schemaType="default"
        initialValues={{ 'source-type': 'google' }}
      />
    );

    const wizzardField = wrapper.find(Wizard).instance();
    expect(wizzardField.state.activeStep).toEqual(1);
    const nextButton = wrapper.find('button').last();
    nextButton.simulate('click');
    expect(wizzardField.state.activeStep).toEqual('google-step');
  });
});
