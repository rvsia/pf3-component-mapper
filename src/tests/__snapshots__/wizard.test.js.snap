// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wizard /> should render Wizard correctly 1`] = `
<Wizard
  FieldProvider={[Function]}
  buttonLabels={
    Object {
      "back": "Back",
      "cancel": "Cancel",
      "next": "Next",
      "submit": "Submit",
    }
  }
  fields={
    Array [
      Object {
        "fields": Array [],
        "name": "step-1",
        "nextStep": "step-2",
        "stepKey": 1,
        "title": "Step 1",
      },
      Object {
        "fields": Array [],
        "name": "step-2",
        "stepKey": "step-2",
        "title": "Step 2",
      },
    ]
  }
  formOptions={
    Object {
      "getState": [Function],
      "onCancel": [MockFunction],
      "onSubmit": [MockFunction],
      "submit": [Function],
      "valid": true,
    }
  }
  inModal={false}
  name="Wizard"
>
  <WizardStep
    FieldProvider={[Function]}
    buttonLabels={
      Object {
        "back": "Back",
        "cancel": "Cancel",
        "next": "Next",
        "submit": "Submit",
      }
    }
    disableBack={true}
    fields={Array []}
    formOptions={
      Object {
        "getState": [Function],
        "handleSubmit": [Function],
        "onCancel": [MockFunction],
        "onSubmit": [MockFunction],
        "submit": [Function],
        "valid": true,
      }
    }
    handleNext={[Function]}
    handlePrev={[Function]}
    name="step-1"
    nextStep="step-2"
    stepKey={1}
    title="Step 1"
  >
    <WizardBody
      className=""
    >
      <ModalBody
        bsClass="modal-body"
        className="wizard-pf-body clearfix"
        componentClass="div"
      >
        <div
          className="wizard-pf-body clearfix modal-body"
        >
          <WizardRow
            className=""
          >
            <section
              className="wizard-pf-row"
            >
              <WizardMain
                className=""
              >
                <div
                  className="wizard-pf-main"
                >
                  <div
                    className="form-horizontal"
                  />
                </div>
              </WizardMain>
            </section>
          </WizardRow>
        </div>
      </ModalBody>
    </WizardBody>
    <WizardStepButtons
      FieldProvider={[Function]}
      buttonLabels={
        Object {
          "back": "Back",
          "cancel": "Cancel",
          "next": "Next",
          "submit": "Submit",
        }
      }
      disableBack={true}
      formOptions={
        Object {
          "getState": [Function],
          "handleSubmit": [Function],
          "onCancel": [MockFunction],
          "onSubmit": [MockFunction],
          "submit": [Function],
          "valid": true,
        }
      }
      handleNext={[Function]}
      handlePrev={[Function]}
      name="step-1"
      nextStep="step-2"
      stepKey={1}
      title="Step 1"
    >
      <WizardFooter
        className=""
      >
        <ModalFooter
          bsClass="modal-footer"
          className="wizard-pf-footer"
          componentClass="div"
        >
          <div
            className="wizard-pf-footer modal-footer"
          >
            <_default
              color="secondary"
              onClick={[MockFunction]}
              style={
                Object {
                  "marginRight": 20,
                }
              }
              type="button"
              variant="contained"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                color="secondary"
                disabled={false}
                onClick={[MockFunction]}
                style={
                  Object {
                    "marginRight": 20,
                  }
                }
                type="button"
                variant="contained"
              >
                <button
                  className="btn btn-default"
                  color="secondary"
                  disabled={false}
                  onClick={[MockFunction]}
                  style={
                    Object {
                      "marginRight": 20,
                    }
                  }
                  type="button"
                  variant="contained"
                >
                  Cancel
                </button>
              </Button>
            </_default>
            <_default
              disabled={true}
              onClick={[Function]}
              type="button"
              variant="contained"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={true}
                onClick={[Function]}
                style={
                  Object {
                    "marginLeft": 3,
                  }
                }
                type="button"
                variant="contained"
              >
                <button
                  className="btn btn-default"
                  disabled={true}
                  onClick={[Function]}
                  style={
                    Object {
                      "marginLeft": 3,
                    }
                  }
                  type="button"
                  variant="contained"
                >
                  <Icon
                    name="angle-left"
                    type="fa"
                  >
                    <FontAwesome
                      name="angle-left"
                    >
                      <span
                        aria-hidden={true}
                        className="fa fa-angle-left"
                      />
                    </FontAwesome>
                  </Icon>
                  Back
                </button>
              </Button>
            </_default>
            <SimpleNext
              FieldProvider={[Function]}
              buttonLabels={
                Object {
                  "back": "Back",
                  "cancel": "Cancel",
                  "next": "Next",
                  "submit": "Submit",
                }
              }
              getState={[Function]}
              handleNext={[Function]}
              next="step-2"
              onCancel={[MockFunction]}
              onSubmit={[MockFunction]}
              submit={[Function]}
              valid={true}
            >
              <_default
                bsStyle="primary"
                onClick={[Function]}
                type="button"
              >
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="primary"
                  disabled={false}
                  onClick={[Function]}
                  style={
                    Object {
                      "marginLeft": 3,
                    }
                  }
                  type="button"
                >
                  <button
                    className="btn btn-primary"
                    disabled={false}
                    onClick={[Function]}
                    style={
                      Object {
                        "marginLeft": 3,
                      }
                    }
                    type="button"
                  >
                    Next
                    <Icon
                      name="angle-right"
                      type="fa"
                    >
                      <FontAwesome
                        name="angle-right"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-angle-right"
                        />
                      </FontAwesome>
                    </Icon>
                  </button>
                </Button>
              </_default>
            </SimpleNext>
          </div>
        </ModalFooter>
      </WizardFooter>
    </WizardStepButtons>
  </WizardStep>
</Wizard>
`;

exports[`<Wizard /> should render Wizard with conditional steps correctly 1`] = `
<Wizard
  FieldProvider={[Function]}
  buttonLabels={
    Object {
      "back": "Back",
      "cancel": "Cancel",
      "next": "Next",
      "submit": "Submit",
    }
  }
  fields={
    Array [
      Object {
        "fields": Array [],
        "name": "step-1",
        "nextStep": Object {
          "stepMapper": Object {
            "step": "step-2",
          },
          "when": "step",
        },
        "stepKey": 1,
        "title": "Step 1",
      },
      Object {
        "fields": Array [],
        "name": "step-2",
        "stepKey": "step-2",
        "title": "Step 2",
      },
    ]
  }
  formOptions={
    Object {
      "getState": [Function],
      "onCancel": [MockFunction],
      "onSubmit": [MockFunction],
      "submit": [Function],
      "valid": true,
    }
  }
  inModal={false}
  name="Wizard"
>
  <WizardStep
    FieldProvider={[Function]}
    buttonLabels={
      Object {
        "back": "Back",
        "cancel": "Cancel",
        "next": "Next",
        "submit": "Submit",
      }
    }
    disableBack={true}
    fields={Array []}
    formOptions={
      Object {
        "getState": [Function],
        "handleSubmit": [Function],
        "onCancel": [MockFunction],
        "onSubmit": [MockFunction],
        "submit": [Function],
        "valid": true,
      }
    }
    handleNext={[Function]}
    handlePrev={[Function]}
    name="step-1"
    nextStep={
      Object {
        "stepMapper": Object {
          "step": "step-2",
        },
        "when": "step",
      }
    }
    stepKey={1}
    title="Step 1"
  >
    <WizardBody
      className=""
    >
      <ModalBody
        bsClass="modal-body"
        className="wizard-pf-body clearfix"
        componentClass="div"
      >
        <div
          className="wizard-pf-body clearfix modal-body"
        >
          <WizardRow
            className=""
          >
            <section
              className="wizard-pf-row"
            >
              <WizardMain
                className=""
              >
                <div
                  className="wizard-pf-main"
                >
                  <div
                    className="form-horizontal"
                  />
                </div>
              </WizardMain>
            </section>
          </WizardRow>
        </div>
      </ModalBody>
    </WizardBody>
    <WizardStepButtons
      FieldProvider={[Function]}
      buttonLabels={
        Object {
          "back": "Back",
          "cancel": "Cancel",
          "next": "Next",
          "submit": "Submit",
        }
      }
      disableBack={true}
      formOptions={
        Object {
          "getState": [Function],
          "handleSubmit": [Function],
          "onCancel": [MockFunction],
          "onSubmit": [MockFunction],
          "submit": [Function],
          "valid": true,
        }
      }
      handleNext={[Function]}
      handlePrev={[Function]}
      name="step-1"
      nextStep={
        Object {
          "stepMapper": Object {
            "step": "step-2",
          },
          "when": "step",
        }
      }
      stepKey={1}
      title="Step 1"
    >
      <WizardFooter
        className=""
      >
        <ModalFooter
          bsClass="modal-footer"
          className="wizard-pf-footer"
          componentClass="div"
        >
          <div
            className="wizard-pf-footer modal-footer"
          >
            <_default
              color="secondary"
              onClick={[MockFunction]}
              style={
                Object {
                  "marginRight": 20,
                }
              }
              type="button"
              variant="contained"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                color="secondary"
                disabled={false}
                onClick={[MockFunction]}
                style={
                  Object {
                    "marginRight": 20,
                  }
                }
                type="button"
                variant="contained"
              >
                <button
                  className="btn btn-default"
                  color="secondary"
                  disabled={false}
                  onClick={[MockFunction]}
                  style={
                    Object {
                      "marginRight": 20,
                    }
                  }
                  type="button"
                  variant="contained"
                >
                  Cancel
                </button>
              </Button>
            </_default>
            <_default
              disabled={true}
              onClick={[Function]}
              type="button"
              variant="contained"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={true}
                onClick={[Function]}
                style={
                  Object {
                    "marginLeft": 3,
                  }
                }
                type="button"
                variant="contained"
              >
                <button
                  className="btn btn-default"
                  disabled={true}
                  onClick={[Function]}
                  style={
                    Object {
                      "marginLeft": 3,
                    }
                  }
                  type="button"
                  variant="contained"
                >
                  <Icon
                    name="angle-left"
                    type="fa"
                  >
                    <FontAwesome
                      name="angle-left"
                    >
                      <span
                        aria-hidden={true}
                        className="fa fa-angle-left"
                      />
                    </FontAwesome>
                  </Icon>
                  Back
                </button>
              </Button>
            </_default>
            <ConditionalNext
              FieldProvider={[Function]}
              buttonLabels={
                Object {
                  "back": "Back",
                  "cancel": "Cancel",
                  "next": "Next",
                  "submit": "Submit",
                }
              }
              getState={[Function]}
              handleNext={[Function]}
              nextStep={
                Object {
                  "stepMapper": Object {
                    "step": "step-2",
                  },
                  "when": "step",
                }
              }
              onCancel={[MockFunction]}
              onSubmit={[MockFunction]}
              submit={[Function]}
              valid={true}
            >
              <MockFieldProvider
                name="step"
                subscription={
                  Object {
                    "value": true,
                  }
                }
              >
                <SimpleNext
                  buttonLabels={
                    Object {
                      "back": "Back",
                      "cancel": "Cancel",
                      "next": "Next",
                      "submit": "Submit",
                    }
                  }
                  getState={[Function]}
                  handleNext={[Function]}
                  onCancel={[MockFunction]}
                  onSubmit={[MockFunction]}
                  submit={[Function]}
                  valid={true}
                >
                  <_default
                    bsStyle="primary"
                    onClick={[Function]}
                    type="button"
                  >
                    <Button
                      active={false}
                      block={false}
                      bsClass="btn"
                      bsStyle="primary"
                      disabled={false}
                      onClick={[Function]}
                      style={
                        Object {
                          "marginLeft": 3,
                        }
                      }
                      type="button"
                    >
                      <button
                        className="btn btn-primary"
                        disabled={false}
                        onClick={[Function]}
                        style={
                          Object {
                            "marginLeft": 3,
                          }
                        }
                        type="button"
                      >
                        Next
                        <Icon
                          name="angle-right"
                          type="fa"
                        >
                          <FontAwesome
                            name="angle-right"
                          >
                            <span
                              aria-hidden={true}
                              className="fa fa-angle-right"
                            />
                          </FontAwesome>
                        </Icon>
                      </button>
                    </Button>
                  </_default>
                </SimpleNext>
              </MockFieldProvider>
            </ConditionalNext>
          </div>
        </ModalFooter>
      </WizardFooter>
    </WizardStepButtons>
  </WizardStep>
</Wizard>
`;

exports[`<Wizard /> should render Wizard with stepsInfo correctly 1`] = `
<Wizard
  FieldProvider={[Function]}
  buttonLabels={
    Object {
      "back": "Back",
      "cancel": "Cancel",
      "next": "Next",
      "submit": "Submit",
    }
  }
  fields={
    Array [
      Object {
        "fields": Array [],
        "name": "step-1",
        "nextStep": "step-2",
        "stepKey": 1,
        "title": "Step 1",
      },
      Object {
        "fields": Array [],
        "name": "step-2",
        "stepKey": "step-2",
        "title": "Step 2",
      },
    ]
  }
  formOptions={
    Object {
      "getState": [Function],
      "onCancel": [MockFunction],
      "onSubmit": [MockFunction],
      "submit": [Function],
      "valid": true,
    }
  }
  inModal={false}
  name="Wizard"
  stepsInfo={
    Array [
      Object {
        "title": "step1",
      },
      Object {
        "title": "step2",
      },
    ]
  }
>
  <WizardSteps
    className=""
    steps={
      Array [
        <WizardStep
          activeStep={1}
          className=""
          label="1"
          onClick={[Function]}
          step={1}
          stepIndex={1}
          title="step1"
        />,
        <WizardStep
          activeStep={1}
          className=""
          label="2"
          onClick={[Function]}
          step={2}
          stepIndex={2}
          title="step2"
        />,
      ]
    }
  >
    <div
      className="wizard-pf-steps"
    >
      <ul
        className="wizard-pf-steps-indicator"
      >
        <WizardStep
          activeStep={1}
          className=""
          key="1"
          label="1"
          onClick={[Function]}
          step={1}
          stepIndex={1}
          title="step1"
        >
          <li
            className="wizard-pf-step active"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              <span
                className="wizard-pf-step-number"
              >
                1
              </span>
              <span
                className="wizard-pf-step-title"
              >
                step1
              </span>
            </a>
          </li>
        </WizardStep>
        <WizardStep
          activeStep={1}
          className=""
          key="2"
          label="2"
          onClick={[Function]}
          step={2}
          stepIndex={2}
          title="step2"
        >
          <li
            className="wizard-pf-step"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              <span
                className="wizard-pf-step-number"
              >
                2
              </span>
              <span
                className="wizard-pf-step-title"
              >
                step2
              </span>
            </a>
          </li>
        </WizardStep>
      </ul>
    </div>
  </WizardSteps>
  <WizardStep
    FieldProvider={[Function]}
    buttonLabels={
      Object {
        "back": "Back",
        "cancel": "Cancel",
        "next": "Next",
        "submit": "Submit",
      }
    }
    disableBack={true}
    fields={Array []}
    formOptions={
      Object {
        "getState": [Function],
        "handleSubmit": [Function],
        "onCancel": [MockFunction],
        "onSubmit": [MockFunction],
        "submit": [Function],
        "valid": true,
      }
    }
    handleNext={[Function]}
    handlePrev={[Function]}
    name="step-1"
    nextStep="step-2"
    stepKey={1}
    title="Step 1"
  >
    <WizardBody
      className=""
    >
      <ModalBody
        bsClass="modal-body"
        className="wizard-pf-body clearfix"
        componentClass="div"
      >
        <div
          className="wizard-pf-body clearfix modal-body"
        >
          <WizardRow
            className=""
          >
            <section
              className="wizard-pf-row"
            >
              <WizardMain
                className=""
              >
                <div
                  className="wizard-pf-main"
                >
                  <div
                    className="form-horizontal"
                  />
                </div>
              </WizardMain>
            </section>
          </WizardRow>
        </div>
      </ModalBody>
    </WizardBody>
    <WizardStepButtons
      FieldProvider={[Function]}
      buttonLabels={
        Object {
          "back": "Back",
          "cancel": "Cancel",
          "next": "Next",
          "submit": "Submit",
        }
      }
      disableBack={true}
      formOptions={
        Object {
          "getState": [Function],
          "handleSubmit": [Function],
          "onCancel": [MockFunction],
          "onSubmit": [MockFunction],
          "submit": [Function],
          "valid": true,
        }
      }
      handleNext={[Function]}
      handlePrev={[Function]}
      name="step-1"
      nextStep="step-2"
      stepKey={1}
      title="Step 1"
    >
      <WizardFooter
        className=""
      >
        <ModalFooter
          bsClass="modal-footer"
          className="wizard-pf-footer"
          componentClass="div"
        >
          <div
            className="wizard-pf-footer modal-footer"
          >
            <_default
              color="secondary"
              onClick={[MockFunction]}
              style={
                Object {
                  "marginRight": 20,
                }
              }
              type="button"
              variant="contained"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                color="secondary"
                disabled={false}
                onClick={[MockFunction]}
                style={
                  Object {
                    "marginRight": 20,
                  }
                }
                type="button"
                variant="contained"
              >
                <button
                  className="btn btn-default"
                  color="secondary"
                  disabled={false}
                  onClick={[MockFunction]}
                  style={
                    Object {
                      "marginRight": 20,
                    }
                  }
                  type="button"
                  variant="contained"
                >
                  Cancel
                </button>
              </Button>
            </_default>
            <_default
              disabled={true}
              onClick={[Function]}
              type="button"
              variant="contained"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={true}
                onClick={[Function]}
                style={
                  Object {
                    "marginLeft": 3,
                  }
                }
                type="button"
                variant="contained"
              >
                <button
                  className="btn btn-default"
                  disabled={true}
                  onClick={[Function]}
                  style={
                    Object {
                      "marginLeft": 3,
                    }
                  }
                  type="button"
                  variant="contained"
                >
                  <Icon
                    name="angle-left"
                    type="fa"
                  >
                    <FontAwesome
                      name="angle-left"
                    >
                      <span
                        aria-hidden={true}
                        className="fa fa-angle-left"
                      />
                    </FontAwesome>
                  </Icon>
                  Back
                </button>
              </Button>
            </_default>
            <SimpleNext
              FieldProvider={[Function]}
              buttonLabels={
                Object {
                  "back": "Back",
                  "cancel": "Cancel",
                  "next": "Next",
                  "submit": "Submit",
                }
              }
              getState={[Function]}
              handleNext={[Function]}
              next="step-2"
              onCancel={[MockFunction]}
              onSubmit={[MockFunction]}
              submit={[Function]}
              valid={true}
            >
              <_default
                bsStyle="primary"
                onClick={[Function]}
                type="button"
              >
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="primary"
                  disabled={false}
                  onClick={[Function]}
                  style={
                    Object {
                      "marginLeft": 3,
                    }
                  }
                  type="button"
                >
                  <button
                    className="btn btn-primary"
                    disabled={false}
                    onClick={[Function]}
                    style={
                      Object {
                        "marginLeft": 3,
                      }
                    }
                    type="button"
                  >
                    Next
                    <Icon
                      name="angle-right"
                      type="fa"
                    >
                      <FontAwesome
                        name="angle-right"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-angle-right"
                        />
                      </FontAwesome>
                    </Icon>
                  </button>
                </Button>
              </_default>
            </SimpleNext>
          </div>
        </ModalFooter>
      </WizardFooter>
    </WizardStepButtons>
  </WizardStep>
</Wizard>
`;

exports[`<Wizard /> should render Wizard with title correctly 1`] = `
<Wizard
  FieldProvider={[Function]}
  buttonLabels={
    Object {
      "back": "Back",
      "cancel": "Cancel",
      "next": "Next",
      "submit": "Submit",
    }
  }
  fields={
    Array [
      Object {
        "fields": Array [],
        "name": "step-1",
        "nextStep": "step-2",
        "stepKey": 1,
        "title": "Step 1",
      },
      Object {
        "fields": Array [],
        "name": "step-2",
        "stepKey": "step-2",
        "title": "Step 2",
      },
    ]
  }
  formOptions={
    Object {
      "getState": [Function],
      "onCancel": [MockFunction],
      "onSubmit": [MockFunction],
      "submit": [Function],
      "valid": true,
    }
  }
  inModal={false}
  name="Wizard"
  title="Wizard title"
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <div
      className="modal-header"
    >
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4"
      >
        <h4
          className="modal-title"
        >
          Wizard title
        </h4>
      </ModalTitle>
    </div>
  </ModalHeader>
  <WizardStep
    FieldProvider={[Function]}
    buttonLabels={
      Object {
        "back": "Back",
        "cancel": "Cancel",
        "next": "Next",
        "submit": "Submit",
      }
    }
    disableBack={true}
    fields={Array []}
    formOptions={
      Object {
        "getState": [Function],
        "handleSubmit": [Function],
        "onCancel": [MockFunction],
        "onSubmit": [MockFunction],
        "submit": [Function],
        "valid": true,
      }
    }
    handleNext={[Function]}
    handlePrev={[Function]}
    name="step-1"
    nextStep="step-2"
    stepKey={1}
    title="Step 1"
  >
    <WizardBody
      className=""
    >
      <ModalBody
        bsClass="modal-body"
        className="wizard-pf-body clearfix"
        componentClass="div"
      >
        <div
          className="wizard-pf-body clearfix modal-body"
        >
          <WizardRow
            className=""
          >
            <section
              className="wizard-pf-row"
            >
              <WizardMain
                className=""
              >
                <div
                  className="wizard-pf-main"
                >
                  <div
                    className="form-horizontal"
                  />
                </div>
              </WizardMain>
            </section>
          </WizardRow>
        </div>
      </ModalBody>
    </WizardBody>
    <WizardStepButtons
      FieldProvider={[Function]}
      buttonLabels={
        Object {
          "back": "Back",
          "cancel": "Cancel",
          "next": "Next",
          "submit": "Submit",
        }
      }
      disableBack={true}
      formOptions={
        Object {
          "getState": [Function],
          "handleSubmit": [Function],
          "onCancel": [MockFunction],
          "onSubmit": [MockFunction],
          "submit": [Function],
          "valid": true,
        }
      }
      handleNext={[Function]}
      handlePrev={[Function]}
      name="step-1"
      nextStep="step-2"
      stepKey={1}
      title="Step 1"
    >
      <WizardFooter
        className=""
      >
        <ModalFooter
          bsClass="modal-footer"
          className="wizard-pf-footer"
          componentClass="div"
        >
          <div
            className="wizard-pf-footer modal-footer"
          >
            <_default
              color="secondary"
              onClick={[MockFunction]}
              style={
                Object {
                  "marginRight": 20,
                }
              }
              type="button"
              variant="contained"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                color="secondary"
                disabled={false}
                onClick={[MockFunction]}
                style={
                  Object {
                    "marginRight": 20,
                  }
                }
                type="button"
                variant="contained"
              >
                <button
                  className="btn btn-default"
                  color="secondary"
                  disabled={false}
                  onClick={[MockFunction]}
                  style={
                    Object {
                      "marginRight": 20,
                    }
                  }
                  type="button"
                  variant="contained"
                >
                  Cancel
                </button>
              </Button>
            </_default>
            <_default
              disabled={true}
              onClick={[Function]}
              type="button"
              variant="contained"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={true}
                onClick={[Function]}
                style={
                  Object {
                    "marginLeft": 3,
                  }
                }
                type="button"
                variant="contained"
              >
                <button
                  className="btn btn-default"
                  disabled={true}
                  onClick={[Function]}
                  style={
                    Object {
                      "marginLeft": 3,
                    }
                  }
                  type="button"
                  variant="contained"
                >
                  <Icon
                    name="angle-left"
                    type="fa"
                  >
                    <FontAwesome
                      name="angle-left"
                    >
                      <span
                        aria-hidden={true}
                        className="fa fa-angle-left"
                      />
                    </FontAwesome>
                  </Icon>
                  Back
                </button>
              </Button>
            </_default>
            <SimpleNext
              FieldProvider={[Function]}
              buttonLabels={
                Object {
                  "back": "Back",
                  "cancel": "Cancel",
                  "next": "Next",
                  "submit": "Submit",
                }
              }
              getState={[Function]}
              handleNext={[Function]}
              next="step-2"
              onCancel={[MockFunction]}
              onSubmit={[MockFunction]}
              submit={[Function]}
              valid={true}
            >
              <_default
                bsStyle="primary"
                onClick={[Function]}
                type="button"
              >
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="primary"
                  disabled={false}
                  onClick={[Function]}
                  style={
                    Object {
                      "marginLeft": 3,
                    }
                  }
                  type="button"
                >
                  <button
                    className="btn btn-primary"
                    disabled={false}
                    onClick={[Function]}
                    style={
                      Object {
                        "marginLeft": 3,
                      }
                    }
                    type="button"
                  >
                    Next
                    <Icon
                      name="angle-right"
                      type="fa"
                    >
                      <FontAwesome
                        name="angle-right"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-angle-right"
                        />
                      </FontAwesome>
                    </Icon>
                  </button>
                </Button>
              </_default>
            </SimpleNext>
          </div>
        </ModalFooter>
      </WizardFooter>
    </WizardStepButtons>
  </WizardStep>
</Wizard>
`;
